.github/workflows/publish-must-gather.yaml

```diff
@@ -0,0 +1,92 @@
+name: "Publish Python Auto-Instrumentation"
+
+on:
+  push:
+    branches: [ main ]
+    tags: [ 'v*' ]
+
+  workflow_dispatch:
+
+env:
+  PLATFORMS: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
+
+jobs:
+  publish:
+    name: Publish must-gather container image
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '~1.22.4'
+
+      - name: Unshallow
+        run: git fetch --prune --unshallow
+
+      - name: Set env vars for the job
+        run: |
+          echo "VERSION_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
+          echo "VERSION=$(git describe --tags | sed 's/^v//')" >> $GITHUB_ENV
+
+      - name: Build the binary for each supported architecture
+        run: |
+          for platform in $(echo $PLATFORMS | tr "," "\n"); do
+            arch=${platform#*/}
+            echo "Building must-gather for $arch"
+            make must-gather ARCH=$arch
+          done
+
+      - name: Docker meta
+        id: docker_meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
```

Comment:
I think we should remove `otel/opentelemetry-operator-must-gather` and publish only to ghcr like we do for other images